lib/ReactViews/Analytics/PolygonParameterEditor.jsx

```diff
@@ -38,7 +38,7 @@ const PolygonParameterEditor = React.createClass({
     },
 
     getValue() {
-        const pointsLongLats = this.props.parameterValues[this.props.parameter.id];
+        const pointsLongLats = this.props.parameterValues[this.props.parameter.id][0];
```

Comment:
this.props.parameterValues[key] data type varies depending on which ParameterEditor has set it.

In InvokeFunction.jsx we have this._parameterValues (https://github.com/TerriaJS/terriajs/blob/rd_wps_fixes/lib/ReactViews/Analytics/InvokeFunction.jsx#L21), which in initializeParameters, goes through and sets any defaults defined for every parameter (https://github.com/TerriaJS/terriajs/blob/rd_wps_fixes/lib/ReactViews/Analytics/InvokeFunction.jsx#L65). So this._parameterValues might look something like:
{"layer" : "some_default_layer", "point": undefined}, where "layer" has come from a LiteralData parameter and "point" has come from a PointGeometry parameter.

this._parameterValues is passed to the ParameterEditor for every type of Parameter (https://github.com/TerriaJS/terriajs/blob/rd_wps_fixes/lib/ReactViews/Analytics/InvokeFunction.jsx#L90), so the this._parameterValues for each ParameterEditor (PolygonParameterEditor.jsx, say) contains a default for the parameters for that editor but also defaults for the other parameters.

It seems every *ParameterEditor.jsx is sticking in whatever data type it wants for parameterValues. That's okay, because each parameter editor is processing that value before it's returned to calling functions, with:
* *Parameter.js's formatValueAsString to display the value in the WPS dialog
* *ParameterEditor.jsx's getValue to retrieve a value which will be used to construct a WPS Execute request: https://github.com/TerriaJS/terriajs/blob/rd_wps_fixes/lib/Models/WebProcessingServiceCatalogFunction.js#L251

Does that make sense?